src/main/groovy/edu/oregonstate/mist/personsapi/mapper/PersonMapper.groovy

```diff
@@ -14,6 +14,17 @@ import java.sql.SQLException
 public class PersonMapper implements ResultSetMapper<PersonObject> {
     PhoneFormatter phoneFormatter = new PhoneFormatter()
     public PersonObject map(int i, ResultSet rs, StatementContext sc) throws SQLException {
+        String rawSSN = rs.getString('SSN')
+        String ssnStatus = 'invalid'
+
+        if (rawSSN && rawSSN.startsWith('VAULT')) {
+            ssnStatus = 'valut'
+        } else if (rawSSN == 'REDACTED' || rawSSN == 'ARCHIVED') {
+            ssnStatus = rawSSN.toLowerCase()
+        } else if (rawSSN && rawSSN.matches(/^(?!0{9})\d{9}$/)) {
+            ssnStatus = 'valid'
+        }
```

Comment:
This might be a little overly paranoid, but would we do something like set `rawSSN` to `null` after the logic for getting `ssnStatus` has completed to make sure this variable's value is never used anywhere else? I guess this wouldn't be necessary if we move the logic to its own private function, since it would be freed from memory when the method scope is exited, right?

Reply:
That's a really good point, I'll think about it more.

Reply:
@ianbrown9475 I found this article and it shows the best practice about how to store sensitive data into variables: https://dev.to/awwsmm/how-to-encrypt-a-password-in-java-42dh

However, think about it more, I think since we only need to know the SSN staus, I should prevent to get the rawSSN in the first place. Therefore, I move the calculation logic to SQL query, so that we'll always get the ssn status for persons.

Reply:
This sounds like the best solution